tests/test_utils.c

```diff
@@ -86,20 +86,20 @@ Test(utils_unit, mmm_ident) {
 
 Test(utils_unit, mmm_rotation) {
   size_t dim = 2;
-  double res[] = {0.0, 0.0, 0.0, 0.0};
-  double a[] = {1.0, 2.0, 3.0, 4.0};
-  double b[] = {0.0, -1.0, 1.0, 0.0};
-  double expected[] = {3.0, 1.0, 4.0, 2.0};
```

Comment:
@jakobbeckmann  this is the line of the test that seems suspect

Reply:
Yeah ok I checked and basically the array equality check from criterion does not seem to work on doubles. The better way to check seems to be:

```c
for(size_t col = 0; col < dim; col++) {
  for(size_t row = 0; row < dim; row++) {
    cr_expect_float_eq(expected[row * dim + col], res[row * dim + col], 0.001, "entry [%ld, %ld] should be correct", row, col);
  }
}
```

Then the test also fails with doubles until you use:

```c
double expected[] = {2.0, -1.0, 4.0, -3.0};
```

Note that `expect` is nicer in this case, since it will not abort execution, therefore you can tell how many values in the matrix are wrong, and not just if one is wrong.

In fact, `expect` is nicer in most cases unless your test breaks if the assertion fails, I might refactor the tests at some point if I have nothing better to do for the project.

Reply:
I am pretty sure there are also other places in the tests where `cr_assert_arr_eq` is used to compare arrays. I will create an issue for test refactoring and take care of this real quick this weekend or maybe even Friday night.